### ISSUE TITLE
Support Binary columns in the CSV writer

### ISSUE THREAD

**Issue by asayers** [apache/arrow-rs#3291]
Trying to write an arrow table containing binary columns using the CSV writer results in the following error:

```
ArrowReadWriteError(CsvError("CSV Writer does not support Binary data type"))
```

Instead, it should print the column somehow:

* simple hex (`format!({:x})`) would be better than nothing
* using [escape_ascii](https://doc.rust-lang.org/std/primitive.slice.html#method.escape_ascii) might be preferable

**Comment by hiltontj** [comment]
`FixedSizeBinary` is encoded in CSV using the default arrow formatter, i.e., as HEX, because at the site where that error is emitted, the check does not match on `FixedSizeBinary`: https://github.com/apache/arrow-rs/blob/dfe0f2627b6e28b49062cc5c6f6174903597bb46/arrow-csv/src/writer.rs#L139-L141

Given https://github.com/apache/arrow-rs/issues/5620, I think it makes sense to encode binary types as HEX, using the arrow formatter.

**Comment by hiltontj** [comment]
This should be closed by #5782 - I mistakenly linked that PR to the wrong issue 🤦
